dandi/cli/base.py

```diff
@@ -30,6 +30,29 @@ def get_metavar(self, param):
         return "N[:M]"
 
 
+class ChoiceList(click.ParamType):
+    name = "choice-list"
+
+    def __init__(self, values):
+        self.values = set(values)
+
+    def convert(self, value, param, ctx):
+        if value is None or isinstance(value, set):
+            return value
+        selected = set()
+        for v in value.split(","):
+            if v == "all":
+                selected = self.values.copy()
+            elif v in self.values:
+                selected.add(v)
+            else:
+                self.fail(f"{v!r}: invalid value", param, ctx)
```

Comment:
I see that it is just a moved code but may be while at it:

```suggestion
                self.fail(f"{v!r}: invalid value. Known are: {', '.join(self.values)}", param, ctx)
```
(would lack `all`) or

```suggestion
                self.fail(f"{v!r}: invalid value. Known are: {self.get_metavar(None)}", param, ctx)
```

to make it easier for people to enter correct values.   Someone fancy could have even used smth like https://github.com/datalad/datalad/blob/HEAD/datalad/utils.py#L2587 to assume that it was a typo and suggest close by choices

Reply:
Done: 04fbfa1